src/engine/enginebuffer.cpp

```diff
@@ -1130,6 +1132,15 @@ void EngineBuffer::process(CSAMPLE* pOutput, const int iBufferSize) {
     }
 #endif
 
+    if (m_pTestSaveBuffer != NULL) {
```

Comment:
I would prefer to add this block in a process() override function. 
This way it dos not pollute the code under test.